bindings/python/docs/source/schemas.rst

```diff
@@ -0,0 +1,86 @@
+.. _schema usage:
+
+Schema Examples
+===============
+
+The following are a few examples of usage of PyMongoArrow Schemas in common situations.
+
+
+Nested Data With Schema
+-----------------------
+
+With aggregate or find methods, you can provide a schema for nested data using the struct object. Note that there can be conflicting
+names in sub-documents compared to their parent documents.
+
+.. code-block:: python
```

Comment:
I think we should adopt https://github.com/adamchainz/blacken-docs to clean up our docs.  

For this PR, that would mean:
- Change the code block type to `code-block:: pycon`
- Use `...` on continuation lines
- Get `blacken-docs` to parse
- Commit the changes

And then, open tickets for ARROW and PYTHON to add `blacken-docs` to our `pre-commit` hooks.

Reply:
This makes sense to me. Done.

Reply:
```
(arrow3.10) ➜  docs git:(ARROW-150) ✗ blacken-docs source/schemas.rst
(arrow3.10) ➜  docs git:(ARROW-150) ✗ git add .
(arrow3.10) ➜  docs git:(ARROW-150) ✗ git commit -m "conform to blacken-docs"
```
`blacken-docs` exits without rewriting.

Reply:
Great!  Can you please open those two tickets?

Reply:
https://jira.mongodb.org/browse/ARROW-154 and https://jira.mongodb.org/browse/PYTHON-3610 !